internal/controller/datadogagent/global/global.go

```diff
@@ -236,22 +236,35 @@ func applyGlobalSettings(logger logr.Logger, manager feature.PodTemplateManagers
 		}
 	}
 
-	// Configure FIPS Agent image or FIPS proxy.
-	if apiutils.BoolValue(config.UseFIPSAgent) {
-		// Add -fips suffix to each container image
-		for i, container := range manager.PodTemplateSpec().Spec.Containers {
-			// Note: if an image override is configured, this image tag will be overwritten
-			manager.PodTemplateSpec().Spec.Containers[i].Image = container.Image + defaulting.FIPSTagSuffix
-		}
-		for i, container := range manager.PodTemplateSpec().Spec.InitContainers {
-			manager.PodTemplateSpec().Spec.InitContainers[i].Image = container.Image + defaulting.FIPSTagSuffix
-		}
-	} else if config.FIPS != nil && apiutils.BoolValue(config.FIPS.Enabled) {
+	// Update images with Global Registry and UseFIPSAgent configurations
+	updateContainerImages(config, manager)
+
+	// Apply FIPS proxy settings - UseFIPSAgent must be false
+	if !*config.UseFIPSAgent && config.FIPS != nil && apiutils.BoolValue(config.FIPS.Enabled) {
 		applyFIPSConfig(logger, manager, dda, resourcesManager)
 	}
 
 }
 
+func updateContainerImages(config *v2alpha1.GlobalConfig, podTemplateManager feature.PodTemplateManagers) {
+	image := &images.Image{}
+	for i, container := range podTemplateManager.PodTemplateSpec().Spec.Containers {
+		image = images.FromString(container.Image)
```

Comment:
nit - `FromString` can be made a method of `Image` and this could be done like
```go
image.FromString(container.Image).
  WithRegistry(..).
  WithFips(..).ToString()
```

Reply:
thanks, partially took the suggestion as i found it easier to reason if it's not a struct method